### PR TITLE
fix: Conan packaging of SDK

### PR DESCRIPTION
**Pull request by BjoernAtBosch**
## Describe your changes

Make the C++ SDK usable as "real" Conan package, e.g. from template

## Issue ticket number and link

AB#16326 (task 16305)

## Checklist - Manual tasks

* [ ] Examples are executing successfully
* [ ] Created/updated unit tests. Code Coverage percentage on new code shall be >= 80%.
* [ ] Created/updated integration tests.
* [ ] Devcontainer can be opened successfully
* [ ] Devcontainer can be opened successfully behind a corporate proxy
* [ ] Devcontainer can be re-built successfully
* [ ] Extended the documentation (e.g. README.md, CONTRIBUTING.md, Velocitas)

### CODE REVIEW

File: conanfile.py [hunk 1]
```diff
@@ -19,43 +19,37 @@
 
 class VehicleAppCppSdkConan(ConanFile):
     name = "vehicle-app-sdk"
-    version = "0.3.3"
+    version = "prep-rel-0.4.0"
```
[the remark this answers is not in the document]
Comment (doosuu) [reply conanfile.py]: Refactored this to be dynamic.

**Review by doosuu — DISMISSED** [review]
LGTM

**Review by MP91 — APPROVED** [review]
LGTM

**Review by MP91 — APPROVED** [review]
lgtm

**Pull request merged**